src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1028 @@
+        if head_mask is not None:
+            probs = probs * head_mask
+        output = torch.einsum("bhsm,bhmk->bhsk", probs, v)  # [batch, heads, sequence, kernel]
+        output = output.transpose(1, 2)  # [batch, sequence, heads, kernel]
+        output = torch.einsum("bshk,hkc->bsc", output, self.o)  # [batch, sequence, hidden]
+
+        if use_cache is True:
+            outputs = (output, present)
+        else:
+            outputs = (output,)
+
+        if output_attentions:
+            outputs += (probs,)
+
+        return outputs
+
+
+class GPTSANJapaneseLayerSelfAttention(nn.Module):
+    """
+    Self Attention and Normalization Unit
+    """
+
+    def __init__(self, config, has_relative_attention_bias=False):
+        super().__init__()
+        self.SelfAttention = GPTSANJapaneseAttention(config)
+        self.norm = nn.LayerNorm(config.d_model, eps=config.layer_norm_epsilon)
+
+    def forward(
+        self,
+        hidden_states: Optional[Tuple[torch.FloatTensor]],
+        layer_past: Optional[Tuple[torch.Tensor]] = None,
+        attention_mask: Optional[torch.FloatTensor] = None,
+        head_mask: Optional[torch.FloatTensor] = None,
+        use_cache: Optional[bool] = False,
+        output_attentions: Optional[bool] = False,
+    ) -> Tuple[Union[torch.Tensor, Tuple[torch.Tensor]], ...]:
+        r"""
+        Self-attention and normalize block.
+
+        Args:
+            hidden_states (`torch.Tensor`) :
+                [num_groups, tokens_per_group, hidden_dim] inputs to send to experts.
+            layer_past (`torch.Tensor`) :
+                past status from previous present output with using `use_cache`.
+            attention_mask (`torch.Tensor`) :
+                apply mask for attention.
+            head_mask (`torch.Tensor`) :
+                apply mask for heads.
+            use_cache (`bool`) :
+                output present key values.
+            output_attentions (`bool`) :
+                output attention probabirities.
+        Returns:
+            Tuple[torch.Tensor[num_groups, tokens_per_group, hidden_dim],...]
+        """
+        atten_out = self.SelfAttention(
+            hidden_states=hidden_states,
+            layer_past=layer_past,
+            attention_mask=attention_mask,
+            head_mask=head_mask,
+            use_cache=use_cache,
+            output_attentions=output_attentions,
+        )
+        attention_output = atten_out[0]
+
+        hidden = hidden_states + self.norm(attention_output)
+
+        outputs = (hidden,) + atten_out[1:]
+
+        return outputs
+
+
+class GPTSANJapaneseBlock(nn.Module):
+    """
+    Self Attention and FFN Unit
+    """
+
+    def __init__(self, config, ext_layer=False):
+        super().__init__()
+        self.SelfAttention = GPTSANJapaneseLayerSelfAttention(config)
+        self.FeedForward = GPTSANJapaneseLayerDenseFF(config) if ext_layer else GPTSANJapaneseLayerSparseFF(config)
```

Comment:
Gain, for class variable, let's use lower case and not Camel. 
```suggestion
        self.self_attn = GPTSANJapaneseLayerSelfAttention(config)
        self.feed_forward = GPTSANJapaneseLayerDenseFF(config) if ext_layer else GPTSANJapaneseLayerSparseFF(config) # or ff
```